File: HashTable/136. Single Number.java
class Solution {
    public int singleNumber(int[] nums) {
        //XOR bitwise is faster
        int result = 0;
        for(int num : nums){
            result ^= num;
        }
        return result;
        //hashsets work but it's slow
        /*
        Set<Integer> hashset = new HashSet<>();
        for (int key : nums) {
            if (hashset.contains(key)) {
                hashset.remove(key);
            }
            else{
                hashset.add(key);
            }
        }
        ArrayList<Integer> result = new ArrayList<>(hashset);

        return result.get(0);
        */
    }
}

File: HashTable/136. Single Number.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        #XOR is faster
        result = 0
        for num in nums:
            result ^= num
        return result

        #hashset works but it's slow
        hashset = set()
        for x in nums:
            if x in hashset:
                hashset.remove(x)
            else:
                hashset.add(x)
        return list(hashset)[0]
        